searxng_extra/update/update_translation_labels.py

```diff
@@ -0,0 +1,133 @@
+
+    :param locale: instance of :py:class:`Locale`
+    :param locale_name: name e.g. 'fr'  or 'pt_BR' (delimiter is *underscore*)
+    """
+
+    native_language, native_territory = _get_locale_descr(locale, locale_name)
+    english_language, english_territory = _get_locale_descr(locale, 'en')
+
+    if native_territory == english_territory:
+        english_territory = None
+
+    if not native_territory and not english_territory:
+        if native_language == english_language:
+            return native_language
+        return native_language + ' (' + english_language + ')'
+
+    result = native_language + ', ' + native_territory + ' (' + english_language
+    if english_territory:
+        return result + ', ' + english_territory + ')'
+    return result + ')'
+
+
+def _get_locale_descr(locale, language_code):
+    language_name = locale.get_language_name(language_code).capitalize()
+    if language_name and ('a' <= language_name[0] <= 'z'):
+        language_name = language_name.capitalize()
+    territory_name = locale.get_territory_name(language_code)
+    return language_name, territory_name
+
+
+def main():
+    """Initialize locales environment of the SearXNG session.
+
+    - monkey patch :py:obj:`flask_babel.get_translations` by :py:obj:`get_translations`
+    - init global names :py:obj:`LOCALE_NAMES`, :py:obj:`RTL_LOCALES`
+    """
+
+    searx_directory = pathlib.Path(__file__).parent.parent.parent / 'searx'
+    translations_filename = searx_directory / 'data/update_translation_labels.json'
+    translations_directory = searx_directory / 'translations'
+
+    for tag, descr in ADDITIONAL_TRANSLATIONS.items():
+        locale = babel.Locale.parse(LOCALE_BEST_MATCH[tag], sep='-')
+        LOCALE_NAMES[tag] = descr
+        if locale.text_direction == 'rtl':
+            RTL_LOCALES.add(tag)
+
+    for tag in LOCALE_BEST_MATCH:
+        descr = LOCALE_NAMES.get(tag)
+        if not descr:
+            locale = babel.Locale.parse(tag, sep='-')
+            LOCALE_NAMES[tag] = get_locale_descr(locale, tag.replace('-', '_'))
+            if locale.text_direction == 'rtl':
+                RTL_LOCALES.add(tag)
+
+    for dirname in sorted(os.listdir(translations_directory)):
+        # Based on https://flask-babel.tkte.ch/_modules/flask_babel.html#Babel.list_translations
+        if not os.path.isdir(os.path.join(translations_directory, dirname, 'LC_MESSAGES')):
+            continue
+        tag = dirname.replace('_', '-')
+        descr = LOCALE_NAMES.get(tag)
+        if not descr:
+            locale = babel.Locale.parse(dirname)
+            LOCALE_NAMES[tag] = get_locale_descr(locale, dirname)
+            if locale.text_direction == 'rtl':
+                RTL_LOCALES.add(tag)
+
+    with open(translations_filename, 'w', encoding='utf-8') as f:
+        content = {
+            "LOCALE_NAMES": LOCALE_NAMES,
+            "RTL_LOCALES": list(RTL_LOCALES),
+            "LOCALE_BEST_MATCH": LOCALE_BEST_MATCH,
```

Comment:
wikipedia uses this variable

Reply:
In the new commit `ADDITIONAL_TRANSLATIONS` and `LOCALE_BEST_MATCH` are back in the `searx.locales`.